src/ApiClient.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\CdtClient;
+
+use Http\Message\MultipartStream\MultipartStreamBuilder;
+use League\Container\Argument\LiteralArgument;
+use League\Container\Container;
+use OpenEuropa\CdtClient\Contract\ApiClientInterface;
+use OpenEuropa\CdtClient\Contract\EndpointInterface;
+use OpenEuropa\CdtClient\Endpoint\MainEndpoint;
+use OpenEuropa\CdtClient\Endpoint\TokenEndpoint;
+use OpenEuropa\CdtClient\Model\Token;
+use OpenEuropa\CdtClient\Traits\TokenAwareTrait;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Psr\Http\Message\UriFactoryInterface;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+
+/**
+ * Class ApiClient
+ *
+ * Provides a centralized client for interacting with the CDT API.
+ * It handles requesting and setting up what is necessary to execute calls to the endpoints.
+ *
+ * @see ApiClientInterface
+ * @see TokenAwareTrait
+ */
+class ApiClient implements ApiClientInterface
```

Comment:
The class does not implement `TokenAwareInterface`, so the IDE will underline things in tests

Reply:
yes lets drop the token trait from here and implement the token getter setters or split up the trait ? up to you

Reply:
I would implement them here, and then see what happens when we add all endpoints. Maybe at that point we'll come up with some idea to avoid a mess in this file